nc-news/src/components/Topics.jsx

```diff
@@ -0,0 +1,25 @@
+import { useParams } from "react-router-dom";
+import { useEffect } from "react";
+import { ArticleCard } from "../index";
+import { fetchAllArticles } from "../../utils/utils";
+
+export default function Topics({ setAllArticles, allArticles }) {
+  const { topic } = useParams();
+
+  useEffect(() => {
+    fetchAllArticles().then((res) => {
+      setAllArticles(res);
+    });
+  }, []);
+
+  return (
+    <section className="articles">
+      <h2>#{topic}</h2>
+      {allArticles.map((article) => {
+        if (article.topic === topic) {
+          return <ArticleCard key={article.article_id} article={article} />;
```

Comment:
This map is handling the filtering of the Articles on the client side. 

_If_ your backend project doesn't have the ability to add the _query_ to get articles filtered by a topic then this is all you can do for now

This is something I would keep mental note of to come back to whenever you have time to expand the backend ( if it's not already able to ) to get the query up and running, and then make use of that as part of your _AllArticles_ component, instead of using a map 👍